### PR TITLE
support vimrc map & noremap  

### PR DESCRIPTION
**Pull request by jjoekoullas**
**What this PR does / why we need it**:
Adds support for `map` and `renoremap` in .vimrc 

**Which issue(s) this PR fixes**
#4403 

**Special notes for your reviewer**:
Had a test failure however it's currently failing on master.

```
  1) Remapper
       leader, c -> closeActiveEditor in visual mode through modehandler:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

1 !== 0
```

### CODE REVIEW

**Comment by J-Fields** [comment]
>Had a test failure however it's currently failing on master.
Hmm, what platform are you on?

File: src/configuration/vimrc.ts [hunk 1]
```diff
@@ -42,33 +42,54 @@ class VimrcImpl {
...
         default:
+          console.warn('Encountered an unrecognized mapping type.');
           return undefined;
       }
     })();
 
-    // Don't override a mapping present in settings.json; those are more specific to VSCodeVim.
-    if (remaps && !remaps.some(r => _.isEqual(r.before, remap!.keyRemapping.before))) {
-      remaps.push(remap.keyRemapping);
+    if (!mappings) {
```
Comment (J-Fields) [inline src/configuration/vimrc.ts]: This shouldn't be necessary - the `forEach` will be a no-op if `mappings` is empty

File: src/configuration/vimrc.ts [hunk 2]
```diff
@@ -42,33 +42,54 @@ class VimrcImpl {
...
    */
   private static addRemapToConfig(config: IConfiguration, remap: IVimrcKeyRemapping): void {
-    const remaps = (() => {
+    const mappings = (() => {
       switch (remap.keyRemappingType) {
+        case 'map':
+          return [
+            config.normalModeKeyBindings,
+            config.visualModeKeyBindings,
+            config.insertModeKeyBindings,
```
Comment (J-Fields) [inline src/configuration/vimrc.ts]: According to [this](https://vim.fandom.com/wiki/Mapping_keys_in_Vim_-_Tutorial_(Part_1)) and a bit of testing, seems `map` doesn't create a mapping for insert/command line modes.

**Comment by jjoekoullas** [comment]
> > Had a test failure however it's currently failing on master.
> 
> Hmm, what platform are you on?

Ubuntu.  I can write up a bug report if you'd like.

File: src/configuration/vimrc.ts (hunk 1 above)
Comment (jjoekoullas) [reply src/configuration/vimrc.ts]: `mappings` could be undefined (not empty) from switch statement.  Also tsc complains that mappings might be null or undefined.  So I think this is necessary to keep the error of an unrecognized mapping type as non-fatal (as it was before).

File: src/configuration/vimrc.ts (hunk 2 above)
Comment (jjoekoullas) [reply src/configuration/vimrc.ts]: Ah, thanks.  Good catch,

Fixed.

File: src/configuration/vimrc.ts (hunk 1 above)
Comment (J-Fields) [reply src/configuration/vimrc.ts]: Oh I see. Could you remove this check and do `mappings?.forEach` instead?

File: src/configuration/vimrc.ts (hunk 2 above)
Comment (J-Fields) [reply src/configuration/vimrc.ts]: Hey @jjoekoullas did you push your changes? I'm still seeing `config.insertModeKeyBindings`

File: src/configuration/vimrc.ts (hunk 1 above)
Comment (jjoekoullas) [reply src/configuration/vimrc.ts]: Done.

File: src/configuration/vimrc.ts (hunk 2 above)
Comment (jjoekoullas) [reply src/configuration/vimrc.ts]: Ah, actually I just missed it.  Fixed now.

**Pull request merged**